x-pack/plugins/encrypted_saved_objects/server/lib/encrypted_saved_objects_audit_logger.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectDescriptor, descriptorToArray } from './encrypted_saved_objects_service';
+
+/**
+ * Represents all audit events the plugin can log.
+ */
+export class EncryptedSavedObjectsAuditLogger {
+  constructor(private readonly auditLogger: any) {}
+
+  public encryptAttributeFailure(attributeName: string, descriptor: SavedObjectDescriptor) {
+    this.auditLogger.log(
+      'encrypt_failure',
+      `Failed to encrypt attribute "${attributeName}" for saved object "[${descriptorToArray(
```

Comment:
__note:__ I can use something like `join(':')` here and below if you think it's better. Both approaches feel more or less equal to me.